app/views/contexts/summary_context.py

```diff
@@ -69,57 +77,40 @@ def summary(self, collapsible):
         }
         return context
 
-    def section_summary(self, current_location):
-        section_id = current_location.section_id
+    def get_list_summaries(self, current_location):
         section = self._schema.get_section(current_location.section_id)
-        list_item_id = current_location.list_item_id
-        list_name = current_location.list_name
-        block = self._schema.get_block(current_location.block_id)
-
-        groups = self.build_groups_for_section(section_id, list_name, list_item_id)
-        context = {
-            'summary': {
-                'groups': groups,
-                'answers_are_editable': True,
-                'collapsible': block.get('collapsible', False),
-                'summary_type': 'SectionSummary',
-            }
-        }
-
-        title = self._schema.get_section(section_id).get('title')
-
-        placeholder_renderer = PlaceholderRenderer(
-            language=self._language,
-            schema=self._schema,
-            answer_store=self._answer_store,
-            metadata=self._metadata,
+        visible_list_collector_blocks = self._schema.get_visible_list_blocks_for_section(
+            section
+        )
+        section_path = self._path_finder.routing_path(
+            section['id'], current_location.list_item_id
         )
+        section_path_block_ids = [location.block_id for location in section_path]
+        list_summaries = []
 
-        if list_item_id:
+        for list_collector_block in visible_list_collector_blocks:
+            add_link_list_name = list_collector_block['for_list']
+            add_link_block_id = list_collector_block['add_block']['id']
 
-            repeating_title = self._schema.get_repeating_title_for_section(section_id)
-            if repeating_title:
-                title = placeholder_renderer.render_placeholder(
-                    repeating_title, list_item_id
+            if list_collector_block['id'] not in section_path_block_ids:
+                driving_question_block = QuestionnaireSchema.get_driving_question_for_section(
+                    section, list_collector_block['for_list']
                 )
 
-        list_collector_blocks = self._schema.get_visible_list_blocks_for_section(
-            section
-        )
-
-        list_summaries = []
+                if driving_question_block:
+                    add_link_list_name = None
+                    add_link_block_id = driving_question_block['id']
```

Comment:
It would be clearer to have an `add_link` variable set to either:
```python
add_link = url_for(
    'questionnaire.block',
    list_name=list_collector_block['for_list'],
    block_id=list_collector_block['add_block']['id'],
)
```
or
```python
add_link = url_for(
    'questionnaire.block',
    block_id= driving_question_block['id'],
)
```